drivers/cache/cache_handlers.c

```diff
@@ -7,30 +7,86 @@
 #include <zephyr/cache.h>
 #include <zephyr/syscall_handler.h>
 
-static inline int z_vrfy_sys_cache_data_all(int op)
+static inline int z_vrfy_sys_cache_data_flush_all(void)
 {
-	return z_impl_sys_cache_data_all(op);
+	return z_impl_sys_cache_data_flush_all();
 }
-#include <syscalls/sys_cache_data_all_mrsh.c>
+#include <syscalls/sys_cache_data_flush_all_mrsh.c>
 
-static inline int z_vrfy_sys_cache_data_range(void *addr, size_t size, int op)
+static inline int z_vrfy_sys_cache_instr_flush_all(void)
+{
+	return z_impl_sys_cache_instr_flush_all();
+}
+#include <syscalls/sys_cache_instr_flush_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_data_invd_all(void)
+{
+	return z_impl_sys_cache_data_invd_all();
+}
+#include <syscalls/sys_cache_data_invd_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_instr_invd_all(void)
+{
+	return z_impl_sys_cache_instr_invd_all();
+}
+#include <syscalls/sys_cache_instr_invd_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_data_flush_and_invd_all(void)
+{
+	return z_impl_sys_cache_data_flush_and_invd_all();
+}
+#include <syscalls/sys_cache_data_flush_and_invd_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_instr_flush_and_invd_all(void)
+{
+	return z_impl_sys_cache_instr_flush_and_invd_all();
+}
+#include <syscalls/sys_cache_instr_flush_and_invd_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_data_flush_range(void *addr, size_t size)
+{
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(addr, size));
+
+	return z_impl_sys_cache_data_flush_range(addr, size);
+}
+#include <syscalls/sys_cache_data_flush_range_mrsh.c>
+
+static inline int z_vrfy_sys_cache_instr_flush_range(void *addr, size_t size)
+{
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(addr, size));
+
+	return z_impl_sys_cache_instr_flush_range(addr, size);
+}
+#include <syscalls/sys_cache_instr_flush_range_mrsh.c>
+
+static inline int z_vrfy_sys_cache_data_invd_range(void *addr, size_t size)
+{
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(addr, size));
+
+	return z_impl_sys_cache_data_invd_range(addr, size);
+}
+#include <syscalls/sys_cache_data_invd_range_mrsh.c>
```

Comment:
Still, I don't fully understand the purpose of exposing the cache management API to userspace - it isn't clear either from the linked issue or from this PR description.

Do we have some real examples why do we implement this?

Reply:
Yeah, I was thinking for example to some userspace app using OpenAMP to exchange data with remote cores using shared memory.

See: https://github.com/OpenAMP/open-amp/blob/80555a314e5cdc92533fec26e48941cd2db60108/lib/virtio/virtqueue.c#L32-L34

Reply:
I cannot really say if this is the only case tbh. At the same time I don't have a good use case to expose the `cache_instr_` syscalls, so I'm going to remove those as well for now.

Reply:
removed